bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BalRuntime.java

```diff
@@ -101,7 +102,7 @@ public void stop() {
         if (!moduleInitialized) {
             throw ErrorHelper.getRuntimeException(ErrorCodes.INVALID_METHOD_CALL, "stop");
         }
-        invokeMethodAsync("$moduleStop", null, PredefinedTypes.TYPE_NULL, "stop", new Object[1]);
+        invokeMethodAsync("$moduleStop", null, PredefinedTypes.TYPE_NULL, "stop", new Scheduler(false), null);
```

Comment:
Why we move scheduler creation from outside module stop?

Reply:
In the old design we created a scheduler inside `$moduleStop` and then invoked `.<stop>` methods of all the dependencies. But in the new design,
- we invoke `.<stop>` of only the current module.
- then we invoke `$moduleStop` methods of immediate dependencies. They will then invoke their `.<stop>` methods and the `$moduleStop` methods of their immediate dependencies.

Since we invoke multiple `$moduleStop` methods, we have to extract the scheduler creation outside of the `$moduleStop` chains. Otherwise we will have duplicate scheduler instances for each module. Therefore we moved the scheduler creation to `run` method in `$_init$SignalListener` class. Which invokes the `$moduleStop` method of the very first module.

A `.class` file wise change comparison can be found [this document](https://docs.google.com/document/d/1yjTv5e38wfYhYoJgmZgb2HH7KXYvOt06c4-GDBsP0wU/edit?usp=sharing).